File: javascript/src/Behavioral/Template_Method/BookTitleInfo.js
(function() {
  var __hasProp = {}.hasOwnProperty,
    __extends = function(child, parent) { for (var key in parent) { if (__hasProp.call(parent, key)) child[key] = parent[key]; } function ctor() { this.constructor = child; } ctor.prototype = parent.prototype; child.prototype = new ctor(); child.__super__ = parent.prototype; return child; };

  define(['Behavioral/Template_Method/TitleInfo'], function(TitleInfo) {
    var BookTitleInfo;
    return BookTitleInfo = (function(_super) {
      __extends(BookTitleInfo, _super);

      BookTitleInfo.prototype.author = null;

      function BookTitleInfo(titleName, author) {
        this.setTitleName(titleName);
        this.setAuthor(author);
      }

      BookTitleInfo.prototype.setAuthor = function(a) {
        return this.author = a;
      };

      BookTitleInfo.prototype.getAuthor = function() {
        return this.author;
      };

      BookTitleInfo.prototype.getTitleBlurb = function() {
        return "Book: " + this.getTitleName() + ", Author: " + this.getAuthor();
      };

      return BookTitleInfo;

    })(TitleInfo);
  });

}).call(this);

File: go/behavioral/templatemethod/BookTitleInfo.go
package templatemethod


//BookTitleInfo - two of three concrete templates

type BookTitleInfo struct {
	titleInfo *TitleInfoImpl
   author string
}

func NewBookTitleInfo(titleName, author string) *BookTitleInfo {
	obj := new(BookTitleInfo)
	obj.titleInfo = new(TitleInfoImpl)
	obj.titleInfo.SetTitleName( titleName )
	obj.author = author
	return obj
}

func (this *BookTitleInfo) getAuthor() string {
	return this.author
}   
   
func (this *BookTitleInfo) GetTitleBlurb() string {
	return "Book: " + this.titleInfo.GetTitleName() + ", Author: " + this.author
}
func (this *BookTitleInfo) GetDvdEncodingRegionInfo() string {
	return this.titleInfo.GetDvdEncodingRegionInfo()
}
